### DISCUSSION TITLE
Multiple Driver Naming

### DISCUSSION THREAD

**Discussion by adrman** [ExistentialAudio/BlackHole#264] [Q&A]
Hi. First, thank you for this project. I want to be able to run multiple instances of the driver, for example, 4 mono instances and 4 2ch instances. I've followed the instructions in the wiki for changing the UID to a different string and then renaming the driver, but so far I'm only able to see one instance of each channel variation in Audio Midi Setup. I noticed that you've added the %ich to append the number of channels and I've tried both keeping it in place and removing it without success. Can you give me any insights on what I'm doing wrong? Again thank you for your work.

**Comment by devinroth** [comment]
Don’t worry about the %ich. Just delete that and make sure everything is unique.

Devin

**Reply by adrman** [reply]
Thanks for the quick reply. I've got to prep for an upcoming session, but I'll give it another shot tomorrow and report back.

**Reply by adrman** [reply]
I'm seeing the multiple instances now Devin. However, each instance is still not reflecting the unique name that I assigned it. Audio Midi setup is showing "BlackHole 1ch 1, 2, etc...", which would be fine, but the final number isn't being passed along to Chrome, Zoom or other apps. Is it necessary to change the UID elsewhere other than the 3 instances described in the wiki?

**Comment by devinroth** [comment]
Change kDataSource_ItemNamePattern and DEVICE_NAME for that.

Devin

**Reply by adrman** [reply]
That did the trick. Thank you again.
